### PR TITLE
Better internal logging system

### PR DESCRIPTION
**Pull request by SkaceKamen**
* Added timestamps
 * Added component identifiers
 * All components now use unified logging system, controller from server

There's a bigger architecture refactoring coming, this is just the first part.

### CODE REVIEW

File: server/src/sqflint.ts [hunk 1]
```diff
@@ -35,6 +37,12 @@ export class SQFLint {
     // Currently running sqflint process
     private childProcess: ChildProcess;
 
+    private logger: Logger
+
+    constructor(context: LoggerContext) {
+        this.logger = context.createLogger('sqflint')
```
Comment (RZSenfo) [inline server/src/sqflint.ts:43]: missing semicolon

File: server/src/server.ts [hunk 2]
```diff
@@ -907,11 +917,11 @@ export class SQFLintServer {
                                     });
                                 }
                             } else {
-                                console.log("Failed to get document", uri);
+                                this.logger.error("Failed to get document", uri);
                             }
                         } catch (e) {
-                            console.log("Failed to load " + uri);
-                            console.log(e);
+                            this.logger.error("Failed to load " + uri);
```
Comment (RZSenfo) [inline server/src/server.ts:923]: might as well stuff this into a single statement?
The logger function allows for multiple args

Comment (SkaceKamen) [reply server/src/server.ts:923]: True

**Review by RZSenfo — APPROVED** [review]
Looking good!
I guess the main issue is the overall structure of the extension, it could need a rework, but i dont know whether thats worth it

**Comment by SkaceKamen** [comment]
Yea, mabye it's a too much effort, for extension that's actually working well right now. But I have this repo on my Github account and I'd like to have something nice here :D

**Pull request merged**